### ISSUE TITLE
How to implement Conditional Masked Autoregressive Flow?

### ISSUE THREAD

**Issue by zhouhanc** [ikostrikov/pytorch-flows#7]
I'm trying to replicate the MNIST and CIFAR-10 experiments used in the paper. If I want to modify the code to add Conditional Masked Autoregressive Flow, which part of the neural network model should I modify?

Thanks for your help!!

**Comment by ikostrikov** [comment]
It's (almost) there (see the cond variables):
https://github.com/ikostrikov/pytorch-flows/blob/master/flows.py#L53

**Comment by zhouhanc** [comment]
That's great. Thanks a lot!

I have one more question. I'm trying to calculate p(x) where x is input data. I know that the forward path of MADE model returns `u` and `log_jacob`, but I'm not sure how to get p(x) out of those two variables. Do you have any idea?

Code snippet I was referring to:
`u, log_jacob = model(x)`

Thanks!

**Comment by ikostrikov** [comment]
It is implemented here (it's a normal density for u + jacobian):
https://github.com/ikostrikov/pytorch-flows/blob/master/main.py#L140

**Comment by ikostrikov** [comment]
Added in https://github.com/ikostrikov/pytorch-flows/commit/0c980f0795baf404355febdf56859b316f74b35f
